pkg/authentication/firebase.go

```diff
@@ -44,3 +49,31 @@ func NewFirebaseWithTokenVerifier(firebaseAuth FirebaseTokenVerifier) Authentica
 		firebaseAuth: firebaseAuth,
 	}
 }
+
+// firebaseRefresher uses the firebase application to refresh the keys used to verify the token signature.
+// The firebase.App has an internal mechanism to avoid repeating this operation for every request.
+type firebaseRefresher struct {
```

Comment:
Is the `Refresher` important enough to be in the name and explain the functionality of this struct? 

Why not go with `firebaseAuth` or `firebaseAuthentication` since the refreshing part is merely an implementation detail (and one that should be transparent at that)?